core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -442,45 +493,51 @@ sealed trait AsyncRequest {
   type Response <: AsyncResponse
   def path: String
   def ctx: Option[Any]
+  def checkInfo: Option[ZkVersionCheck]
+}
+
+case class ZkVersionCheck(checkPath: String, expectedZkVersion: Int) {
+  def checkOp = Op.check(checkPath, expectedZkVersion)
```

Comment:
Can you add the return type to the signature of `checkOp()`?

Reply:
Done.